compositor_pipeline/src/queue/audio_queue.rs

```diff
@@ -192,7 +192,14 @@ impl AudioQueueInput {
             })
             .collect::<Vec<InputSamples>>();
 
-        self.drop_old_samples(pts_range.1, queue_start);
+        // Drop all batches older than `end_pts`. Entire batch (all samples inside) has to be older.
+        while self
```

Comment:
given the assumption that calling code has to have it's cleaner to inline that code